src/averageAges.js

```diff
@@ -52,7 +56,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (!onlyWithSon && person.mother)
+    || (onlyWithSon
+        && person.mother
+        && person.sex === 'm')
+  );
+
+  const mothersAndChildren = children.map(child => {
+    return {
+      mom: people.find(mom => mom.name === child.mother),
+      child: child,
+    };
+  }).filter(momAndChild => momAndChild.mom !== undefined);
```

Comment:
```suggestion
  }).filter(momAndChild => momAndChild.mom);
```
перетворення типів і так поверне true якщо там буде мама, можна не писати !== undefined